### ISSUE TITLE
Cave of Crystals

### ISSUE THREAD

**Issue by Killer66** [shalzuth/BraveHaxvius#85]
The bot is unable to complete CoC stages. Thoughts?

**Comment by lupier** [comment]
Very easy - try to use DataExtractor. Wait until it's finished their job. Then recompile client. That's all
Also remember to validate RST, MST, version values.

**Comment by Killer66** [comment]
What? I think you did not understand the problem. Have you tried to run any Cave of Crystals 3?

**Comment by lupier** [comment]
Sure, all of them are OK. I think you dont understand problem on your side.
example in log:
Cave of Crystals (Support) 3
Sending : Initialize...
	Request done
Sending : GetUserInfo...
	Request done
Sending : MissionWaveStart...
	Request done
XP : 16900
Gil : 3140
Enemy dropped item : Support Alcryst : 6
Enemy dropped item : Support Heavicryst : 3
Enemy dropped item : Support Milcryst : 2
Sending : MissionEnd...
Sending : Initialize...
	Request done
Sending : MissionEnd...
	Request done

**Comment by Killer66** [comment]
Hum.... strange, mine get a error on "Sending : MissionEnd..."

**Comment by lupier** [comment]
"error" is an abstraction. Use visual studio debugger to get error location exactly where it is. Also use local proxy 127.0.0.1 with PacketDecoder to locate any collisions in data

**Comment by Killer66** [comment]
Error on line 111 of networking.cs
var msg = Text.Texts.First(t => t.Key == errorMsg.ToString()).Value;

**Comment by Killer66** [comment]
errorMsg: {SERVER_MSG_158}

**Comment by lupier** [comment]
158 = Please log in again
Check your RST, MST, Ver. values.
app = 1043
rst = 0
mst = 1340
build = ver.2.8.1
You can always check this values in packetdecoder with telephone tuned via proxy

**Comment by lupier** [comment]
Just compare your traffic from client and phone. This is not a "rocket science" :)

**Comment by Killer66** [comment]
These are exactly the same values I have here.

**Comment by lupier** [comment]
make sure your url for php calls is `https://v270lapisdtz9oirblf.gumi.sg/lapisProd/app/php/gme/actionSymbol/`
if this one is already same - compare all decrypted JSON nodes in packet decoder. That's all what you need for debugging purpose

**Comment by Killer66** [comment]
The URL is the same. Tried another instance of the BraveHaxvius solution and worked well, even with very outdated values on MST and APP versions. Thanks for the help anyway.

**Comment by robidk** [comment]
Hi @Killer66 , where did you get the other instance from? I am having same problem with my version. Have updated data, checked version etc but get an Item with same key already exists error.

**Comment by robidk** [comment]
This is the error I am getting, ,in case someone can tell me how to fix it?
![capture](https://user-images.githubusercontent.com/33748956/41823889-1e6d0d42-77ff-11e8-9576-da7fc542bd0a.JPG)

**Comment by Killer66** [comment]
Yeah, that's the same error i'm getting. Just picked a backup file with a friend. However, now even runing only airship is giving me the same error here and there. I'm thinking the problem may be some instability in the server, so I will stop using the bot for a while.

**Comment by robidk** [comment]
@lupier Hi, do you think you can provide a copy of your files? I am not good enough with programming to find the exact problem on my current files :(
Everytime I do a data extract I need to edit some of the equipment and challenge entries as they have double "" in the description which does not allow me to recompile without removing these, am not sure if maybe these is why I am getting the above error? 
I can do TMR farming by catching above error and restarting the missions, but am having trouble with completing the raids and the CoC stages using the bot.

**Comment by lupier** [comment]
You can escape invalid chars with \ 
before: Kill Kanshou with "Bakuyas Vengence""
after: Kill Kanshou with `\"Bakuyas Vengence\""`
This is only one problem (as I remember) in whole challenges.
p.s. always wait till data extractor finished it's task. Do not close it (~10 minutes for example)

**Comment by robidk** [comment]
looks like doing the above instead of just removing the extra "" has resolved the issue! Thanks lupier.
